numpy/core/arrayprint.py

```diff
@@ -1258,7 +1260,7 @@ def dtype_short_repr(dtype):
     """
     # handle these separately so they don't give garbage like str256
     if issubclass(dtype.type, flexible):
-        if dtype.names:
+        if dtype.names is not None:
```

Comment:
There's the wider implication that this is not a good test for structured types, although since it fails only in case that don't matter, perhaps not worth investing time into.